.github/workflows/test.yml

```diff
@@ -44,6 +44,7 @@ jobs:
         include:
           - python: '3.9'
             django: 'Django>=4.1,<4.2'
+            parallel: '--parallel=1'
```

Comment:
it may be worth doing something like https://github.com/django/django/pull/16808/files to ensure we get accurate coverage, while still preserving parallelism

Reply:
Hmm, not sure what the `concurrency` in Coverage's option is for, but we're already running `coverage` with `--parallel-mode` and combining the report, which was necessary to get the full report across different DB setups:

https://github.com/wagtail/wagtail/blob/c220662eed53a936994d63e0195fe0b6441962ef/.github/workflows/test.yml#L426

I'll look into it, thanks

Reply:
running the Django tests in parallel would result in incomplete coverage. The PR I linked to has the magic incantation to make it work properly (via `concurrency = multiprocessing,thread`)

Reply:
Am I right in thinking `--parallel=1` means "run one process in parallel", i.e. "not parallel"? :-) Seems like this could be enormously misleading to anyone reading this. Could we perhaps use `parallel: ''` or arrange things so that this can be omitted entirely for non-parallel runs? Failing that, a comment just saying `# not parallel` would help.

Reply:
Yeah, on newer Django versions there is `--parallel=auto` which is similar to `--parallel` and can help distinguish between `1` and `auto`, but otherwise it does get confusing. I intentionally kept that in so that it's easier for us to turn parallel on/off, but perhaps it's not worth it. I'll try `parallel: ''` and if it doesn't work, I'll just omit it entirely.

Reply:
Looking at https://docs.djangoproject.com/en/4.2/ref/django-admin/#cmdoption-test-parallel

> If you want to use [pdb](https://docs.python.org/3/library/pdb.html#module-pdb) while debugging tests, you must disable parallel execution (--parallel=1). You’ll see something like bdb.BdbQuit if you don’t.

so `--parallel=1` does indeed mean "not parallel"

Good catch!

Reply:
It’s in the PR description 😉

Reply:
tl;dr - maybe some additional comments as to why we don't run them in parallel would help 6 months down the line :P

long version:
Well, yes. I understand the bit around ES tests not being run in parallel. But the tests against the latest stable Django, or sqlite/mysql have it too.

> [...] keep using a single process for some of them in case the traceback is unreadable, which can be the case when running tests in parallel.

you've added `tblib` to the testing dependencies, so that should no longer be the case?

Reply:
From my experience, even installing `tblib` doesn't help sometimes, especially when the error occurs during template rendering :(

See https://code.djangoproject.com/ticket/29023

Anyway, I didn't try `parallel: ''` as I think it might be confusing, so I removed it for cases where we don't run the tests in parallel. I've also added comments explaining the config. See c9fd6b8.